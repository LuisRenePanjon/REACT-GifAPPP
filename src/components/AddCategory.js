import React, { useState } from 'react';
import PropTypes from 'prop-types';
export const AddCategory = ( {setCategories } ) => {
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = ( events ) => {
        setInputValue( events.target.value );
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputValue.trim().length > 2){
            setCategories( categories => [inputValue , ...categories]);
            setInputValue( '' );
        }
    };

    return (
        
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input
                type="text" 
                placeholder="Ingrese una categoria"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
        
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
