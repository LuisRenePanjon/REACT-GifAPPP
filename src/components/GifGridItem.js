import React from 'react'
import PropTypes from 'prop-types';
export const GifGridItem = ({id, title , url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <p>{ title }</p>
            <img src= {url} alt={ title }/>
        </div>
    );
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url  : PropTypes.string.isRequired,
}
