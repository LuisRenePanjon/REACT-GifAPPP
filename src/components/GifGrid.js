// import React, { useState, useEffect } from 'react'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

import { useFechGifs } from "../hooks/useFetchGifs"

export const GifGrid = ( { category } ) => {
    

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs( category)
    //         .then( setImages);
    // }, [ category ]);
    
    const { data: gifs, loading } = useFechGifs( { category } );

    return (
    <>
        <h3 className="card animate__animated animate__fadeIn">{ category }</h3>
        { loading && <p>Loading!!!</p> }
        <div className="card-grid">
            {
                gifs.map(gif => {
                    return (
                        <GifGridItem 
                            key = { gif.id }
                            { ...gif }
                        />
                    );
                })
            }
        </div>
    </>
    )
}
