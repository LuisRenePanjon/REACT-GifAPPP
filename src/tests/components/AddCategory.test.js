import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Test in <AddCategory/>', () => {
    const setCategories = () => {};
    
    test('should correctly display the component', () => {
        const wrapper =shallow(<AddCategory setCategories = {setCategories}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
