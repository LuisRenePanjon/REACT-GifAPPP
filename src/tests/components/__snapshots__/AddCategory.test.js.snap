// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AddCategory/> should correctly display the component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Ingrese una categoria"
    type="text"
    value=""
  />
</form>
`;
